### PR TITLE
Add Gitleaks and pre-commit to the codespace container definition

### PR DESCRIPTION
**Pull request by yotommy**
### Description

Add [Gitleaks](https://github.com/gitleaks/gitleaks/blob/master/README.md) and [`pre-commit`](https://pre-commit.com/) to the codespace container.

Since there is no official distribution for the codespace add-ons:

* Install `pre-commit` via `pip`
* Use `gitleaks-docker` which fetches gitleaks from Dockerhub

These are prerequisites to configuring a pre-commit hook for secret detection.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [ ] Created unit and/or browser tests which fail without the change (if possible)
- [ ] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

#### Database evolutions

Read the guidelines [here](https://github.com/civiform/civiform/wiki/Database#writing-database-evolutions)

- [ ] Assigned two reviewers
- [ ] Guarded against already existing resources using `IF NOT EXISTS` and `IF EXISTS`
- [ ] Downs created to undo changes in Ups

#### User visible changes

- [ ] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings)
  - [ ] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [ ] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [ ] Manually tested at 200% size
- [ ] Manually evaluated tab order

#### New Features

- [ ] Add new FeatureFlag env vars to `server/conf/helper/feature-flags.conf`
- [ ] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [ ] Wrote browser tests with the feature flag off and on, etc.

### Instructions for manual testing

If instructions are needed for manual testing by reviewers, include them here.

### Issue(s) this completes

Relates to #7099

### CODE REVIEW

**Comment by yotommy** [comment]
* **#7422** <a href="https://app.graphite.dev/github/pr/civiform/civiform/7422?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yotommy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -24,7 +24,9 @@
       "installGradle": "false"
     },
     "ghcr.io/devcontainers/features/node:1": {},
-    "ghcr.io/devcontainers/features/python:1": {}
+    "ghcr.io/devcontainers/features/python:1": {},
+    "ghcr.io/devcontainers-contrib/features/pre-commit:2": {},
```
Comment (dkatzz) [inline .devcontainer/devcontainer.json:28]: **suggestion:** @avaleske just made some changes to remove contrib features - could we install these in the onCreateCommand from [this PR](https://github.com/civiform/civiform/pull/7415) instead?

Comment (yotommy) [reply .devcontainer/devcontainer.json]: Thanks for the suggestion. Revised the approach and updated the PR description.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+- repo: https://github.com/gitleaks/gitleaks
+  rev: v8.18.2
+  hooks:
+    - id: gitleaks-docker
```
Comment (dkatzz) [inline .pre-commit-config.yaml:7]: **question:** Just confirming is this what does the installation?

**Review by dkatzz — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (yotommy) [reply .pre-commit-config.yaml:7]: I can't quite tell what is happening under the hood of `pre-commit`, but nothing more is required to be able to run `gitleaks`:

```bash
$ pre-commit 
[WARNING] Unstaged files detected.
[INFO] Stashing unstaged files to /home/vscode/.cache/pre-commit/patch1715609212-4181.
Detect hardcoded secrets.................................................Passed
[INFO] Restored changes from /home/vscode/.cache/pre-commit/patch1715609212-4181.
```

**Pull request merged**